### PR TITLE
Feature/enable proguard for tests

### PR DESCRIPTION
**Pull request by originx**

### CODE REVIEW

File: DBPrefs/build.gradle [hunk 1]
```diff
@@ -11,19 +11,17 @@ android {
         targetSdkVersion defaultTargetSdkVersion
         versionCode 1
         versionName "2.0"
-
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner" // For Instrumentation Unit testing
+        consumerProguardFile('proguard-rules.pro')
```
Comment (hannesa2) [inline DBPrefs/build.gradle]: This was a real missing feature. Thank you

Comment (hannesa2) [reply DBPrefs/build.gradle]: I didn't think about tests

File: DBPrefs/proguard-rules.pro [hunk 2]
```diff
@@ -13,4 +13,34 @@
...
+
+# Gson specific classes
+-dontwarn sun.misc.**
+-keep class sun.misc.Unsafe { *; }
+-dontnote sun.misc.Unsafe
+#-keep class com.google.gson.stream.** { *; }
+
+# Prevent proguard from stripping interface information from TypeAdapterFactory,
+# JsonSerializer, JsonDeserializer instances (so they can be used in @JsonAdapter)
+-keep class * implements com.google.gson.TypeAdapterFactory
```
Comment (hannesa2) [inline DBPrefs/proguard-rules.pro]: in my point of view, this should be delivered by gson

Comment (originx) [reply DBPrefs/proguard-rules.pro]: completely agree but gson is not packed as aar so it cannot be packaged

**Pull request merged**